### PR TITLE
Add yamllint to the linting workflow; Add linting instructions to the README.

### PR DESCRIPTION
**Pull request by kirkrodrigues**
# Description

#194 added yamllint as a linter but in the midst of pending PRs, we held off on adding it to the GitHub workflow. This PR adds it and documents how to run our linting tools.

As part of creating a single task to run all linting, we moved the `clang-format` linting from inside the core component to the top-level of the repo. This is so we can have a single venv for all the linting tools, outside of the main build directory which changes/gets clobbered frequently.

# Validation performed

* Validated the linting workflow succeeds.
* Validated the linting tasks succeed.
* Validated the package build tasks work.

### CODE REVIEW

File: Taskfile.yml [hunk 1]
```diff
@@ -140,6 +141,66 @@ tasks:
       vars:
         COMPONENT: "{{.TASK}}"
 
+  lint-check:
+    deps: ["lint-venv"]
```
Comment (junhaoliao) [inline Taskfile.yml:145]: Is this `deps[]` removable given all the tasks that `link-check` calls has specified the same dependency?

Comment (junhaoliao) [reply Taskfile.yml:145]: Same question with the `deps[]` in task `lint-fix`.

**Review by junhaoliao — DISMISSED** [review]
LGTM. Just have some questions

File: Taskfile.yml (hunk 1 above)
Comment (kirkrodrigues) [reply Taskfile.yml]: Good catch! Fixed.

**Review by junhaoliao — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**